### ISSUE TITLE
Support webpack, jspm, and other bundlers

### ISSUE THREAD

**Issue by pilwon** [Famous/engine#152]
`browserify` is the only bundler that works out of box because of its dependency library, [glslify](https://github.com/stackgl/glslify), that uses browserify transforms. Browserify is great and all, but many of us already have our own favorite bundlers.

Please support `webpack`, `jspm`, and other popular bundlers by either dropping `glslify` from dependencies or adding `npm prepublish` script to pre-run `glslify`.

(Here's a [temporary workaround](https://gist.github.com/pilwon/1d054ed00350f5fea401) I'm currently using. The same script should work for all other bundlers too.)

**Comment by dcsan** [comment]
this dependency also seems to be preventing us bundling for meteor with famono as linked above.

**Comment by trusktr** [comment]
@dcsan I' working on this for Meteor for this purpose: https://github.com/trusktr/rocket-module

**Comment by trusktr** [comment]
:+1: to supporting multiple bundlers!

**Comment by trusktr** [comment]
Here's glsl loaders for webpack: [glsl-loader](https://www.npmjs.com/package/shader-loader) and [glsl-loader](https://www.npmjs.com/package/glsl-loader)

**Comment by pilwon** [comment]
@trusktr These loaders do not help because they don't support `glslify` syntax famous uses. (ex: `#pragma glslify: ...`) There's an official webpack loader for that as well ([glslify-loader](https://github.com/stackgl/glslify-loader)) but famous is currently using an outdated version that is not compatible with it.

**Comment by dcsan** [comment]
@trusktr Gadi cohen was able to hack this into usability apparently.

https://github.com/raix/Meteor-famono/issues/112#issuecomment-105047685

**Comment by trusktr** [comment]
@pilwon Ah, I see. Well, Mixed Mode is in Beta, so hopefully we can have this solved before Famo.us gets out of Beta! Libraries should ideally be bundler agnostic.

**Comment by lightblade** [comment]
Is there any reason why we can't use the standard ES6 module?

**Comment by trusktr** [comment]
@lightblade I'd like to see this all in ES6 Modules too. That'd be great. It should be an easy PR to make...

**Comment by pilwon** [comment]
:+1: to standard ES6 module

**Comment by michaelobriena** [comment]
@pilwon While I love ES6, I don't think the transpolation needed for a project today is a dependency we want at this time.  Using the module system and the class syntax is totally cool in my book but because we work in such a performance constrained environment I don't think we can accept features into the platform where it is not obvious what the exact performance implications are from a straight source code read.

**Comment by pilwon** [comment]
@michaelobriena I agree ES6 support out of box is not necessary and we should probably avoid it to keep the library performant. (It just popped out of my wish list someone commented above :stuck_out_tongue:)

Famous becoming bundler agnostic is all I hope at this point.

**Comment by michaelobriena** [comment]
@pilwon Agree completely

**Comment by lightblade** [comment]
That's the thing about ES6 modules. You can transpire ES6 modules to any other module system, but not the other way around. Heard of anything that can transpile from CommonJS/AMD to ES6 modules?

I think it's best for framework and library authors to write and even distribute in ES6 module format and leave the module transpiling to application programmers.

**Comment by michaelobriena** [comment]
@lightblade I appreciate your opinion and it is something that we will definitely consider.  I think at this time the benefit does not outweigh the problem of creating the illusion that we are using ES6 and having contributors writing code that cannot make it into the platform because they are using too many experimental features of ES6.

That said, this is all subject to change and should enough people want this, we can always migrate.

**Comment by pilwon** [comment]
I really feel we need a separate issue for this ES6 discussion as it is not the only solution to the bundler agnostic feature request this issue originally focuses on.

**Comment by FarhadG** [comment]
We will be discussing this issue this week and hope to have a conclusion very soon.

@michaelobriena @redwoodfavorite

**Comment by alexanderGugel** [comment]
We're probably just going to include the pre-bundled shader.
Something along those lines:

```javascript
var glslify = require('glslify');
var after = require('after');
var fs = require('fs');

var next = after(2, write);

var shaders = {
    vertex: null,
    fragment: null
};

function write() {
    var src = '\'use strict\'; module.exports = ' + JSON.stringify(shaders) + ';';
    fs.writeFile('./webgl-shaders/index.js', src, function(err) {
        if (err) throw err;
    });
}

glslify.bundle('./webgl-shaders/FragmentShader.glsl', null, function(err, source, files) {
    if (err) throw err;
    shaders.fragment = source;
    next();
});

glslify.bundle('./webgl-shaders/VertexShader.glsl', null, function(err, source, files) {
    if (err) throw err;
    shaders.vertex = source;
    next();
});
```

**Comment by TheAlphaNerd** [comment]
@alexanderGugel would this be some sort of prepublish script?

**Comment by alexanderGugel** [comment]
@TheAlphaNerd Yes.

**Comment by adnan-wahab** [comment]
FIXED

**Comment by adnan-wahab** [comment]
We updated glslify to the latest version, which should now run in webpack.

**Comment by hornos** [comment]
I have the same issue with jspm, glslify version 2.x has a 'process.binding is not supported' error in the graceful-fs module. If you change the dependency of Famous/engine back to glslify 1.6.0 it works. It also works with Famous/engine 0.5.2.

**Comment by polyclick** [comment]
@hornos I have the exact same issue with `glslify` and `jspm` (I don't use Famous). 
I've created an issue here: https://github.com/jspm/jspm-cli/issues/1148
